src/irmin-pack/inode.ml

```diff
@@ -755,6 +755,208 @@ struct
       in
       Irmin.Type.map I.t ~pre_hash (fun v -> { find = niet; v }) (fun t -> t.v)
 
+    module type DATA_FORMAT = sig
```

Comment:
The intent of the `inode_intf` file is to avoid duplicating module types like this. You should be able to define it once at the top-level there and alias it twice.